kubetest2/kubetest2-gke/deployer/deployer.go

```diff
@@ -0,0 +1,683 @@
+
+	return flags
+}
+
+// assert that deployer implements types.Deployer
+var _ types.Deployer = &deployer{}
+
+func (d *deployer) Build() error {
+	return fmt.Errorf("build is not implemented for GKE deployer")
+}
+
+// Deployer implementation methods below
+func (d *deployer) Up() error {
+	if err := d.verifyFlags(); err != nil {
+		return err
+	}
+	if err := d.prepareGcpIfNeeded(); err != nil {
+		return err
+	}
+
+	// Create network if it doesn't exist.
+	if runWithNoOutput(exec.Command("gcloud", "compute", "networks", "describe", d.network,
+		"--project="+d.project,
+		"--format=value(name)")) != nil {
+		// Assume error implies non-existent.
+		log.Printf("Couldn't describe network '%s', assuming it doesn't exist and creating it", d.network)
+		if err := runWithOutput(exec.Command("gcloud", "compute", "networks", "create", d.network,
+			"--project="+d.project,
+			"--subnet-mode=auto")); err != nil {
+			return err
+		}
+	}
+	// Create a custom subnet in that network if it was asked for.
+	if d.customSubnet != "" {
+		customSubnetFields := strings.Fields(d.customSubnet)
+		createSubnetCommand := []string{"compute", "networks", "subnets", "create"}
+		createSubnetCommand = append(createSubnetCommand, "--project="+d.project, "--network="+d.network)
+		createSubnetCommand = append(createSubnetCommand, customSubnetFields...)
+		if err := runWithOutput(exec.Command("gcloud", createSubnetCommand...)); err != nil {
+			return err
+		}
+		d.subnetwork = customSubnetFields[0]
+		d.subnetworkRegion = customSubnetFields[1]
+	}
+
+	shape, err := d.shape()
+	if err != nil {
+		return err
+	}
+	def := shape[defaultPool]
+	loc, err := d.location()
+	if err != nil {
+		return err
+	}
+	args := make([]string, len(d.createCommand()))
+	copy(args, d.createCommand())
+	args = append(args,
+		"--project="+d.project,
+		loc,
+		"--machine-type="+def.MachineType,
+		"--image-type="+d.image,
+		"--num-nodes="+strconv.Itoa(def.Nodes),
+		"--network="+d.network,
+	)
+	log.Printf("Line 286")
+
+	args = append(args, def.ExtraArgs...)
+	if strings.ToUpper(d.image) == "CUSTOM" {
+		args = append(args, "--image-family="+d.imageFamily)
+		args = append(args, "--image-project="+d.imageProject)
+	}
+	if d.subnetwork != "" {
+		args = append(args, "--subnetwork="+d.subnetwork)
+	}
+	if d.additionalZones != "" {
+		args = append(args, "--additional-zones="+d.additionalZones)
+		if err := os.Setenv("MULTIZONE", "true"); err != nil {
+			return fmt.Errorf("error setting MULTIZONE env variable: %v", err)
+		}
+
+	}
+	if d.nodeLocations != "" {
+		args = append(args, "--node-locations="+d.nodeLocations)
+		numNodeLocations := strings.Split(d.nodeLocations, ",")
+		if len(numNodeLocations) > 1 {
+			if err := os.Setenv("MULTIZONE", "true"); err != nil {
+				return fmt.Errorf("error setting MULTIZONE env variable: %v", err)
+			}
+		}
+	}
+
+	fmt.Printf("Environment: %v", os.Environ())
+
+	// TODO(zmerlynn): The version should be plumbed through Extract
```

Comment:
this new TODO should probably be you :-)

Reply:
Fair enough.